content/en/blog/news/2022-GSoC-2022-Supply-Chain-Security-Final-Report.md

```diff
@@ -0,0 +1,97 @@
+| ----------------------------------------------------------- | --------------------------------------------------------------------------------------------------- |
+| https://github.com/jenkins-x/jx/pull/8312                   | Install syft using github action and generate SBOM with goreleaser                                  |
+| https://github.com/jenkins-x/jx3-pipeline-catalog/pull/1166 | Add Syft Installation to JX3 pipeline Catalog so it can be used to generate SBOMs                   |
+| https://github.com/jenkins-x/jx3-pipeline-catalog/pull/1173 | Add Oras Steps to support pushing SBOMs as OCI artifacts                                            |
+| https://github.com/jenkins-x-plugins/jx-pipeline/pull/490   | Use Syft installation step in jx-pipeline from jx3-pipelines-catalog                                |
+| https://github.com/jenkins-x-plugins/jx-pipeline/pull/491   | Use Oras bushing SBOMs step from jx3-pipeline-catalog                                               |
+| https://github.com/jenkins-x/jx-docs/pull/3643              | Add documentation for supporting sbom generation and storing for other jx components and our users  |
+| https://github.com/jenkins-x/jx/pull/8351                   | Applying SBOM management to JX                                                                      |
+| https://github.com/jenkins-x/jx3-pipeline-catalog/pull/1190 | Add grype tasks for SBOM vulnerability scanning to jx3-pipeline-catalog                             |
+| https://github.com/cdfoundation/presentations/pull/44       | Add materials for my talk with the CDF Supply Chain Securtiy SIG about SBOMs with Jenkins X         |
+
+There is also this issue to apply all steps on our active JX repositories [here](https://github.com/jenkins-x/jx/issues/8348) (WIP as there are many repositories to be updated) and the list of PRs are referenced in the issue (except those of JX because they included testing and fixing PRs at the beginning).
+
+### Signing Jenkins X artifacts
+
+Description:
+
+Jenkins X is a collection of many components and tools that are used to provide the end-to-end CI/CD solution. Those components are used by our users and are also used by Jenkins X itself. It is important to sign those components to ensure the integrity of the components and the supply chain security of Jenkins X.
+
+Implementation:
+
+There are a set of tools developed by [sigstore](https://sigstore.dev/) to sign and verify artifacts. The most used tool in my implementation is [cosign](https://github.com/sigstore/cosign). I first started with signing the Jenkins X CLI with keys stored in github secrets and then moved to signing the Jenkins X docker images. After that, I switched to use cosign [keyless signing](https://github.com/sigstore/cosign/blob/main/KEYLESS.md) (an experimental feature using other Sigstore components to save the effort by key management ) The list of PRs for this work are:
```

Comment:
> an experimental feature using other Sigstore components to save the effort by key management

an experimental feature using other Sigstore components (rekor and fulcio) to save end users from the hassles of key (mis)management.

Reply:
Done